lib/m2ts/m2ts.js

```diff
@@ -342,10 +330,27 @@ ElementaryStream = function() {
       parsePes(packetData, event);
 
       stream.size = 0;
-
+      if (!stream.enabled) {
+        return;
```

Comment:
You can save a lot of time building `event.data` if you move this condition to the top of the function and set `stream.data` to an empty array before returning.